### PR TITLE
az-actions

### PR DESCRIPTION
**Pull request by markweitzel**
First pass at adding the "actions" API pattern.

### CODE REVIEW

File: azure/Guidelines.md [hunk 1]
```diff
@@ -524,7 +537,7 @@ While removing a value from an enum is a breaking change, adding an enum can be
 }
 ```
 
-Always model an enum as a string unless you are positive that the symbol set will **NEVER** change over time.
+:ballot_box_with_check: **DO** slways model an ```enum``` as a string unless you are positive that the symbol set will **NEVER** change over time.
```
Comment (mkistler) [inline azure/Guidelines.md]:
```suggestion
:ballot_box_with_check: **DO** model an ```enum``` as a string unless you are positive that the symbol set will **NEVER** change over time.
```

File: azure/Guidelines.md [hunk 2]
```diff
@@ -474,8 +472,23 @@ Overall, this is a very brittle design that leads to a poor developer experience
 ## Common API Patterns
 
 ### Performing an Action
+The REST specificaiton is used to model the state of a resource, and is primarily intended to handle CRUD (Create, Read, Update, Delete) operations. However, many services have the requirement to perform an action on a resource, e.g. getting the thumbnail of an image, sending an SMS message. To perform an action on a resource, we introduce the concept of a "controller resource," or simply "controller."
```
Comment (mkistler) [inline azure/Guidelines.md:475]: I think the notion of a "controller resource" is confusing.  These are _not_ "resources" in the way we generally use this term in REST APIs -- you cannot "create" or "delete" or "get" them.  Why not simply call these what they are: "actions"?
> To perform an action on a resource, issue a request using the resource URL with a single additional path segment with the name of the action.

**Review by mkistler — CHANGES_REQUESTED** [review]
I think "controller resources" are overly complicated and confusing.  Let's simply our guidance by just describing "actions".

File: azure/Guidelines.md (hunk 2 above)
Comment (markweitzel) [reply azure/Guidelines.md:475]: +1. I like your simplified version *much* better and will include it. In this case, I think I pulled too much from the existing literature.

**Pull request merged**